docs/building-from-source.md

```diff
@@ -0,0 +1,69 @@
+# C\#
+
+**Build dependencies:**
+
+* .NET 7 SDK
+
+> .NET 7 is needed only to build from source; the compiled binaries support at minimum .NET 5.
+
+## Build
+
+Building TileDB-CSharp from source can be done using the following commands in a [cloned repository](https://github.com/TileDB-Inc/TileDB-CSharp).
+
+```bash
+$ cd TileDB-CSharp/sources/TileDB.CSharp/
+$ dotnet build -c Release
+```
+
+As a final build step, you can verify your installation by running unit tests
+
+```bash
+$ cd ../../tests/TileDB.CSharp.Test/
+$ dotnet test -c Release
+```
+
+## Development
+
+To help get started using TileDB-CSharp, you can run the [TileDB.CSharp.Example](https://github.com/TileDB-Inc/TileDB-CSharp/tree/main/examples/TileDB.CSharp.Example) project:
+
+```bash
+$ cd TileDB-CSharp/examples/TileDB.CSharp.Example/
+$ dotnet run --framework net5.0
```

Comment:
The examples now target a single framework (and I don't anticipate this to change again).

```suggestion
$ dotnet run
```